ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js

```diff
@@ -17,8 +25,24 @@ const ConfirmPageContainerSummary = (props) => {
     origin,
     hideTitle,
     image,
+    transactionType,
+    toAddress,
   } = props;
 
+  const [showNicknamePopovers, setShowNicknamePopovers] = useState(false);
+  const t = useI18nContext();
+  const { toName, isTrusted } = useAddressDetails(toAddress);
```

Comment:
I think the `toAddress` passed here may be the `recipient`, not the contract the user is interacting with? If that's the case the `isTrusted` value here seems to be used in a way in this component that conflates the trust we are conferring to the `recipient` with the `contract` itself.

As a second point, I'm not sure we ever want to suggest that a user fully trust a `contract`, even in the case where they've added the `contract` as a contact, because we have no way of verifying that the method name is in fact what it claims to be.
 cc @danjm @Gudahtt @holantonela

Reply:
Here for instance we are showing the **recipient** address in the spot where I would expect the **contract** address.

<img width="501" alt="Screen Shot 2022-02-25 at 9 18 22 AM" src="https://user-images.githubusercontent.com/34557516/155750134-60a25cd7-c6e4-4996-b5c2-fb7b9beec474.png">

Reply:
@adonesky1 : as I understand the sender is always the current user, in case of contract interaction `contract` to toAddress.

Reply:
For instance on test-app I created this contract and now I see it as recipient:

![Screenshot 2022-02-25 at 11 17 56 PM](https://user-images.githubusercontent.com/2182307/155762877-7ecbc4f6-bd05-4214-9bbb-8b433abedd13.png)
[Medical Re-imbursement.pdf](https://github.com/MetaMask/metamask-extension/files/8143182/Medical.Re-imbursement.pdf)

![Screenshot 2022-02-25 at 11 17 35 PM](https://user-images.githubusercontent.com/2182307/155762861-95861c4b-0592-4439-bb65-800c37d316e3.png)

Reply:
I don’t follow. I’m saying that here you’re representing the recipients address as if it were the contract address.

Reply:
This image is taken on your PR, and the recipient address/contact/identity name is shown instead of the contract address next to the method name. The contract address must go by a different variable name by the time it reaches this component.

Reply:
@adonesky1 
For transaction of type `TRANSACTION_TYPES.CONTRACT_INTERACTION` contract is actually the recipient. My point is that we need to remove `TRANSACTION_TYPES.TRANSFER` or `TRANSACTION_TYPES.TRANSFER_FROM` from this category. For. those transactions contract address is not available.

Reply:
Yes the contract is the actual recipient, my point is that in this case what we are assigning to the variable toAddress is not the contract. This is semantic choice we make differently in different contexts.  Here we want to pass along the contract address and use that variable here instead

Reply:
So I think I realized why we're missing eachother on this. So in the case of a contract interactions where there is no recipient (most things other than `transfer` and `transferFrom`) the `toAddress` is set to the contract address. I think we should probably identify the contract address (as its own prop, so that it doesn't get conflated with recipient when its a tranfer type contract interaction) many components up and pass it along as a separate prop and just always use that for this particular use regardless of the type of contract interaction. Does that make sense?

Reply:
Opened a PR against this PR here: https://github.com/MetaMask/metamask-extension/pull/13776

Reply:
Hey @adonesky1 : the scope of this PR was contract interactions where method name was derived from `data` in the transaction I would prefer to limit code changes to that. Lets create a task for other transaction type and work on it separately.

Reply:
Yep I reverted those already @adonesky1